code/modules/antagonists/gang/gang_actions.dm

```diff
@@ -0,0 +1,87 @@
+/datum/action/innate/gang
+	icon_icon = 'icons/mob/actions/actions_gang.dmi'
+	background_icon_state = "bg_gang"
+	check_flags = AB_CHECK_CONSCIOUS
+
+/datum/action/innate/gang/invitation
+	name = "Invite to Gang"
+	desc = "Invite someone nearby to join your criminal organization. You have a maximum limit on members, choose wisely."
+	button_icon_state = "gang_invite"
+
+/datum/action/innate/gang/invitation/Activate()
+	var/datum/antagonist/gang/boss/boss = owner.mind.has_antag_datum(/datum/antagonist/gang/boss)
+	if(!boss)
+		return
+	var/datum/team/gang/gang = boss.gang
+	if(!gang)
+		return
+	if(gang.members.len >= gang.max_members)
+		to_chat(owner, "<span class='warning'>We can't support more than [gang.max_members] members, bribery and coercion can still make people cooperative.</span>")
+		return
+
+	var/mob/living/carbon/targets = list()
+	for(var/mob/living/carbon/M in get_hearers_in_view(6, owner))
+		if(!M.mind || !M.client || M.stat == DEAD || M == owner || (M.mind in gang.members)) //Needs to have a mind, a client, be not dead and isn't us. Client not included for testing purposes
+			continue
+		targets += M
+	if(!length(targets))
+		return
+
+	var/offer = tgui_input_text(owner, "Please choose what to say for your invitation.", "Gang Invitation", "")
+	if(!offer || !IsAvailable())
+		return
+
+	if(CHAT_FILTER_CHECK(offer))
+		to_chat(usr, "<span class='warning'>You cannot send a message that contains a word prohibited in IC chat!</span>")
+		return
```

Comment:
Hmmm, I think it wouldn't be a bad idea to log this? Just a, [owner] tried to create a gang invite with a filtered word

Reply:
It doesn't do this for any Chat Filter Check in the code, it'd be weird to have just this one be special

Reply:
Got it, nvm then!